scripts/formats.js

```diff
@@ -1,33 +1,113 @@
+const fs = require('fs');
+const path = require('path');
+const SD_scssFormat = require('../node_modules/style-dictionary/lib/common/formats')['scss/variables'];
+
 /**
  * StyleDictionary custom formats
  * https://amzn.github.io/style-dictionary/formats_and_templates
  */
 
+// load and keep breakpoint values in memory
+// for responsive var value output
+const bpProperties = JSON.parse(
+		fs.readFileSync(
+			path.join(__dirname, '..', 'properties', 'layout', 'breakpoints.json')
+		)
+	);
+
+// get media query string from size keyword
+// (e.g. 's', 'm', 'l')
+const getMQBySize = (size) =>
+	`only screen and (min-width: ${bpProperties.layout.breakpoint[size].value})`;
+
+// generated content warning for top of files
 const dateHeader = () =>
 	  '/**\n' +
 		' * Do not edit directly\n' +
 		' * Generated on ' + new Date() + '\n' +
 		' */\n\n';
 
+
+// commonJS module format
+// (!) "color" category only
+//
+// adds every dictionary property to exports
 const commonJS = {
 	name: 'javascript/commonJS',
-	formatter: (dictionary, platform) => dateHeader() + dictionary
+	formatter: (dictionary) => dateHeader() + dictionary
 		.allProperties
+		.filter(prop => prop.attributes.category === "color")
 		.map(prop => `exports.${prop.name} = '${prop.value}';`)
 		.join('\n')
 };
 
+// CSS Custom properties format
+//
+// There is some extra complexity here due to support
+// for responsive custom properties. A "responsive"
+// custom property has a different value at medium
+// and large breakpoints. These properties are assigned
+// a default value in `:root` with overrides scoped
+// to media queries.
+//
+// A "standard" custom property has only one value,
+// and can be assigned in `:root` just once.
 const customProperties = {
 	name: 'css/customProperties',
-	formatter: (dictionary, platform) => dateHeader() +
-		'\n:root {\n' +
-		dictionary
-			.allProperties
-			.map(p => `\t--${p.name}: ${p.value};`)
-			.join('\n') +
-		'\n}'
+	formatter: (dictionary) => {
+		const { allProperties } = dictionary;
+		const lineTab = '\n\t';
+		const isResponsiveProp = (p) => p.path.includes('responsive');
+		const toCSSRule = (prop, size) => size ?
+			`${prop.name}: ${prop.value[size]};`
+			: `${prop.name}: ${prop.value};`;
+
+		return dateHeader() + `
+/* Default values */
+:root {
+	${allProperties
+		.filter(p => !isResponsiveProp(p))
+		.map(p => toCSSRule(p))
+		.join(lineTab)
+	}
+	${allProperties
+		.filter(p => isResponsiveProp(p))
+		.map(p => toCSSRule(p, 'default'))
+		.join(lineTab)
+	}
+}
+
+/* Medium breakpoint overrides */
```

Comment:
I don't love the repetition for values at each breakpoint, but I can't recommend a better way to do it that's as sturdy or clear as the repetition.

Reply:
Using `.filter(p => isResponsiveProp(p))` within media queries will ensure that we're only repeating the custom property definitions that we need.

Non-responsive custom properties will only be defined once in the first `:root` selector